test/integration/consul-container/libs/agent/container.go

```diff
@@ -199,20 +204,24 @@ func (c *consulContainerNode) Exec(ctx context.Context, cmd []string) (int, erro
 	return c.container.Exec(ctx, cmd)
 }
 
+// Upgrade terminates a running container and create a new one using the provided config.
+// The upgraded node will
+//   - use the same node name and the data dir as the old version node
 func (c *consulContainerNode) Upgrade(ctx context.Context, config Config, index int) error {
-	pc, err := readSomeConfigFileFields(config.JSON)
+	_, err := readSomeConfigFileFields(config.JSON)
 	if err != nil {
 		return err
 	}
 
-	consulType := "client"
-	if pc.Server {
-		consulType = "server"
-	}
-	name := utils.RandName(fmt.Sprintf("%s-consul-%s-%d", pc.Datacenter, consulType, index))
+	// NOTE: keep this as we may want to add an option to change the node name in certain scenario
+	// consulType := "client"
+	// if pc.Server {
+	// 	consulType = "server"
+	// }
+	// name := utils.RandName(fmt.Sprintf("%s-consul-%s-%d", pc.Datacenter, consulType, c.id))
```

Comment:
IMHO we just delete all this, including readSomeConfigFileFields, to keep it cleaner.